pennylane/ops/functions/assert_valid.py

```diff
@@ -67,12 +70,14 @@ def _check_decomposition(op, skip_wire_mapping):
         # Check that mapping wires transitions to the decomposition
         wire_map = {w: ascii_lowercase[i] for i, w in enumerate(op.wires)}
         mapped_op = op.map_wires(wire_map)
-        mapped_decomp = mapped_op.decomposition()
-        orig_decomp = op.decomposition()
-        for mapped_op, orig_op in zip(mapped_decomp, orig_decomp):
-            assert (
-                mapped_op.wires == qml.map_wires(orig_op, wire_map).wires
-            ), "Operators in decomposition of wire-mapped operator must have mapped wires."
+        # I don't think this is a good solution, try to think of a better one
```

Comment:
Why not? :thinking: I'm surprised we did not run into problems before with non-decomposable ops?

Reply:
In general we never get to this with non-decomposable ops, because the entire block only runs if `op.has_decomposition`. This is specifically to skip cases where `op.has_decomposition == True`, but if we map the wires, then `mapped_op.has_decomposition == False`.

This is here because `qml.MultiControlledX(wires=[0, 1])` becomes `CNOT` when we call its `map_wires` method. So you can't check that's mapped wires carry over to its decomposition, because it decomposes and maps at the same time. This is sort of a work-around to that problem instead of solving the problem.

Reply:
But I can also just replace this with a more informative comment clarifying what's happening and leave as-is

Reply:
I would be in favor of this, and we should address the "mapping wires simplifies to CNOT" issue separately :)